src/operator/subgraph/dnnl/dnnl_pow_mul_scalar_property.h

```diff
@@ -0,0 +1,148 @@
+#define MXNET_OPERATOR_SUBGRAPH_DNNL_DNNL_POW_MUL_SCALAR_PROPERTY_H_
+#if MXNET_USE_ONEDNN == 1
+
+#include <map>
+#include <string>
+#include <vector>
+
+#include "operator/subgraph/common.h"
+#include "operator/tensor/elemwise_binary_scalar_op.h"
+#include "dnnl_subgraph_base-inl.h"
+
+namespace mxnet {
+namespace op {
+
+class SgDNNLPowMulScalarSelector : public SubgraphSelectorV2 {
+ private:
+  std::vector<const BiDirectedNode*> matched_list_;
+  SelectStatus status_;
+
+ public:
+  bool Select(const BiDirectedNode& seed_node,
+              const std::shared_ptr<NodeAttr>& node_attr) override {
+    if (seed_node.node->op() == Op::Get("_npi_power_scalar")) {
+      matched_list_.clear();
+      matched_list_.emplace_back(&seed_node);
+      status_ = kStart;
+      return true;
+    }
+    return false;
+  }
+
+  bool SelectInput(const BiDirectedNode& n, const BiDirectedNode& input_node) override {
+    return false;
+  }
+
+  bool SelectOutput(const BiDirectedNode& n, const BiDirectedNode& output_node) override {
+    const nnvm::Node* raw_power_scalar_node = n.node;
+    const nnvm::Node* raw_next_node         = output_node.node;
+    if (raw_power_scalar_node->op() && raw_power_scalar_node->op()->name == "_npi_power_scalar") {
+      if (raw_next_node->op() && status_ == kStart &&
+          raw_next_node->op()->name == "_npi_multiply_scalar") {
+        status_ = kSuccess;
+        return true;
+      } else {
+        status_ = kFail;
+        return false;
+      }
+    }
+
+    if (matched_list_.back() != &n) {
+      if (std::find(matched_list_.begin(), matched_list_.end(), &n) != matched_list_.end()) {
+        while (matched_list_.back() != &n) {
+          matched_list_.pop_back();
+        }
+      }
+      status_ = kSuccess;
+      return false;
+    }
+
+    return false;
+  }
+
+  void Reset() override {
+    CHECK_GE(matched_list_.size(), 1);
+    auto new_selector = SgDNNLPowMulScalarSelector();
+    new_selector.Select(*matched_list_[0], nullptr);
+    *this = new_selector;
+  }
+};
+
+class SgDNNLPowMulScalarProperty : public SubgraphProperty {
+ public:
+  SgDNNLPowMulScalarProperty() {}
+
+  static SubgraphPropertyPtr Create() {
+    static const std::string& name = "DNNL PowMulScalar optimization pass";
+    auto property                  = std::make_shared<SgDNNLPowMulScalarProperty>();
+    property->SetAttr<std::string>("property_name", name);
+    property->SetAttr<bool>("inference_only", true);
+    if (dmlc::GetEnv("MXNET_DISABLE_ONEDNN_POW_MUL_SCALAR_OPT", 0)) {
+      property->SetAttr<bool>("disable", true);
+    }
+    return property;
+  }
+
+  nnvm::ObjectPtr CreateSubgraphNode(const nnvm::Symbol& sym,
+                                     const int subgraph_id = 0) const override {
+    nnvm::ObjectPtr n = nnvm::Node::Create();
+
+    std::ostringstream node_name;
+    node_name << "sg_dnnl_pow_mul_scalar_" << std::to_string(subgraph_id);
```

Comment:
you are not using node_name anywhere, so `n->attrs.name = sg_dnnl_pow_mul_scalar_" + std::to_string(subgraph_id)` will be enough